core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -99,7 +106,16 @@ public Ticket getTicket(final String ticketId) {
                 val ticketAgeSeconds = getTicketAgeSeconds(ticket);
                 LOGGER.debug("Ticket [{}] has expired according to policy [{}] after [{}] seconds and [{}] uses and will be removed from the ticket registry",
                     ticketId, ticket.getExpirationPolicy().getName(), ticketAgeSeconds, ticket.getCountOfUses());
-                deleteSingleTicket(ticket);
+                if (ticket instanceof final TicketGrantingTicket tgt) {
+                    val clientInfo = ClientInfoHolder.getClientInfo();
+                    applicationContext.publishEvent(new CasRequestSingleLogoutEvent(this, tgt, clientInfo));
+                    applicationContext.publishEvent(new CasTicketGrantingTicketDestroyedEvent(this, tgt, clientInfo));
+                }
+                try {
+                    deleteTicket(ticket);
```

Comment:
You should probably return `deleteTicket() > 0`?

Reply:
I don't think so. It should always `return false;` for this use case of an expired ticket, whether the deletion actually succeeds or not.

Reply:
I don't follow. Why? You're deleting a ticket that CAS might see as expired. Why should the result be ignored?

Reply:
I see what you mean, thank you.